src/modules/expression-evaluator/src/evaluateExpression.ts

```diff
@@ -76,16 +79,45 @@ export default async function evaluateExpression(
           const property = childrenResolved[0].property
           return extractNode(inputObject, property)
         } catch {
-          return "Can't resolve object"
+          throw new Error("Can't resolve object")
+        }
+
+      case 'API':
+        let urlWithQuery, returnNode
+        try {
+          const url = childrenResolved[0]
+          const queryFields = childrenResolved[1]
+          const queryValues = childrenResolved.slice(2, queryFields.length + 2)
+          returnNode =
+            childrenResolved[queryFields.length + 2] && childrenResolved[queryFields.length + 2]
```

Comment:
Hmm, I think I thought that it would thrown an error if the index referred to a position that doesn't exist (it does in other languages), so I was checking it existed before assigning it. Never mind, I've simplified it to this now, and the tests still all pass.

`returnNode = childrenResolved[queryFields.length + 2]`